src/routes/boot.ts

```diff
@@ -359,14 +366,26 @@ const anonymousBoot = async (
     middleware ? middleware(con, req, res) : {},
     getAnonymousFirstVisit(req.trackingId),
   ]);
+  const isPreOnboardingV2 = firstVisit
+    ? new Date(firstVisit) < onboardingV2Requirement
+    : false;
+
+  const anonymousFlags = adjustAnonymousFlags(flags, [
+    {
+      replacement: 'control',
+      feature: 'onboarding_v2',
+      onCheckValidity: (value) => value !== 'control' && isPreOnboardingV2,
```

Comment:
@idoshamun just one mention for this experiment that it won't show to x% of selected numbers as we further down it here.
Just wanted to mention it for alignment.